Cleanup formatting and minor code changes

ViToni opened the pull request:
pfSense Ticket #5783
https://redmine.pfsense.org/issues/5783

see:
https://github.com/pfsense/FreeBSD-ports/commit/28cdacad47051a02e7866093634de3326242d21e

ViToni committed:
Adding support for fbsd 9/10 due to changes in struct ip.

Original patch submitted by: yuri@rivera.ru

ViToni committed:
Minor change in wording of logging message.

pali commented:
Hi! Please add proper description into commit message and do not add external links... Also maybe you should squash them into one commit.

ViToni commented:
Could you please check again.

ViToni committed:
Replace call to setpgrp with setpgid

Linux and FreeBSD use a different signature for setpgrp.
Both support setpgid which replaces the now obsolete setpgrp and has the same signature on both systems.

pali commented on src/ifvc.c:
this is broken code

pali commented on src/ifvc.c:
Please fix this part and also fix your email address in your git commit.

ViToni commented on src/ifvc.c:
Fixed the formatting related to this change.

The email adress is the same as used by changes done inside the GitHub UI. Which one is prefered for consistency (GitHub email or private email)?

pali commented on src/ifvc.c:
Null pointer dereference is still not fixed. Also rebase patch on top of branch.

Email address must be valid. Does not matter if private or public... Just valid address on which people can reach you (e.g. in case somebody would need to update your patch).

ViToni commented:
Rewrote all email addresses so that only one email is used for changes by my 
Removed the code in question. The puill request consists now mainly of cleanups, chnage of incompatible code und removal of code duplicates.
Especially the code formatting should ease further patches.

ViToni commented on src/ifvc.c:
Maybe you could explain the problem in the above code to me, because my C is quite rusty and I clearly fail to see it.

pali commented on src/ifvc.c:
Null pointer dereference:

allowednet = ...
if (allowednet == NULL) my_log(...);
allowednet->...

pali commented:
"This branch has conflicts that must be resolved"

And please do different things in different commits/pull requests.
Also include description into commit message. People should understand it without need to open external links...

ViToni commented on src/ifvc.c:
I believe code above avoids NULL pointer dereference because logging an error via 

> `if (allowednet == NULL) my_log(LOG_ERR, 0, "Out of memory !");`

leads to 'src/syslog.c' and there it exits after logging:

> `if( Severity <= LOG_ERR )
>         exit( -1 );
> `
